cli/util/audit-bin.go

```diff
@@ -73,9 +75,29 @@ func auditBin(ctx *cli.Context) error {
 		cursor string
 		curOID oid.ID
 		f      = object.NewSearchFilters()
+		wg     sync.WaitGroup
+		tasks  = make(chan func() error)
+		errs   = make(chan error, numWorkers)
 	)
 	f.AddFilter(blockAttr, strconv.FormatUint(curH, 10), object.MatchNumGE)
 
+	for range numWorkers {
+		wg.Add(1)
+		go func() {
+			for f := range tasks {
+				err := f()
+				if err != nil {
+					errs <- err
+					break
+				}
+			}
+		}()
+	}
+	defer func() {
+		close(tasks)
+		wg.Wait()
```

Comment:
We have `fmt.Fprintln(ctx.App.Writer, "Audit is completed.")` line in the end of this function, move it to this `defer`. Otherwise this message will be printed to the user once all tasks are submitted, and then the command may look hanging.